smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/PackageManifestGenerator.kt

```diff
@@ -34,16 +34,20 @@ class PackageManifestGenerator(val ctx: ProtocolGenerator.GenerationContext) {
                     it.getProperty("url", String::class.java).getOrNull() != null ||
                         it.getProperty("scope", String::class.java).getOrNull() != null
                 }
-                val dependenciesByURL = externalDependencies.distinctBy {
-                    it.getProperty("url", String::class.java).getOrNull()
-                        ?: "${it.getProperty("scope", String::class.java).get()}.${it.packageName}"
-                }
+                val dependenciesByURL = externalDependencies
+                    .distinctBy {
+                        it.getProperty("url", String::class.java).getOrNull()
+                            ?: "${it.getProperty("scope", String::class.java).get()}.${it.packageName}"
+                    }
+                    .sorted()
```

Comment:
Both `dependenciesByURL` and `dependenciesByTarget` are `List<SymbolDependency>`.

The `SymbolDependency` `compareTo()` implementation sorts by `packageName` when the `dependencyType` is the same between dependencies.  (Swift doesn't ever set `dependencyType`, and it sets `packageName` to match the Swift published library name.)
https://github.com/smithy-lang/smithy/blob/main/smithy-codegen-core/src/main/java/software/amazon/smithy/codegen/core/SymbolDependency.java#L243

So, these lists get sorted by package name, which in our case is the Swift library name.